nuxeo-drive-core/src/test/java/org/nuxeo/drive/fixtures/GroupChangesTestSuite.java

```diff
@@ -0,0 +1,346 @@
+ * (C) Copyright 2017 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Antoine Taillefer <ataillefer@nuxeo.com>
+ */
+package org.nuxeo.drive.fixtures;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.Serializable;
+import java.util.Collections;
+import java.util.List;
+
+import javax.inject.Inject;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.nuxeo.drive.listener.NuxeoDriveGroupUpdateListener;
+import org.nuxeo.drive.service.FileSystemChangeFinder;
+import org.nuxeo.drive.service.FileSystemItemChange;
+import org.nuxeo.drive.service.NuxeoDriveEvents;
+import org.nuxeo.ecm.core.api.CoreInstance;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.CoreSessionService;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.security.ACE;
+import org.nuxeo.ecm.core.api.security.SecurityConstants;
+import org.nuxeo.ecm.platform.usermanager.UserManager;
+
+/**
+ * Tests the detection of {@link NuxeoDriveEvents#SECURITY_UPDATED_EVENT} events in the audit logs by the
+ * {@link FileSystemChangeFinder} implementations when group changes occur.
+ *
+ * @see NuxeoDriveGroupUpdateListener
+ * @since 9.2
+ */
+public class GroupChangesTestSuite extends AbstractChangeFinderTestCase {
+
+    private static final Log log = LogFactory.getLog(GroupChangesTestSuite.class);
+
+    @Inject
+    protected UserManager userManager;
+
+    @Inject
+    protected CoreSessionService coreSessionService;
+
+    protected CoreSession userSession;
+
+    @Override
+    @Before
+    public void init() throws Exception {
+        lastEventLogId = 0;
+        lastSyncActiveRootDefinitions = "";
+
+        DocumentModel user = userManager.getBareUserModel();
+        user.setPropertyValue("user:username", "user");
+        userManager.createUser(user);
+
+        List<String> members = Collections.singletonList("user");
+        createGroup("group1", members, null);
+        createGroup("group2", members, null);
+        createGroup("parentGroup", null, Collections.singletonList("group1"));
+        createGroup("grandParentGroup", null, Collections.singletonList("parentGroup"));
```

Comment:
Detail: `Arrays.asList` is shorter to write. And in Java 9 we'll even have `List.of` but we're not there yet...